### DISCUSSION TITLE
I want to be part of this organization Google Summer of Code GSOC 2023

### DISCUSSION THREAD

**Discussion by marinicadenisandrei** [wagtail/gsoc#8] [General]
Basic details
_____________________________________________________________________________________
My name is Marinica Denis Andrei, I live in Bucharest, Romania, Eastern European Time (GMT+2). As a bachelor's degree, I graduated from the Polytechnic University of Bucharest, Faculty of Industrial Engineering and Robotics, "Robotics and Computer Science" department, the final grade obtained being 9.5/10, having as a diploma project the realization of a robotic cell for depalletizing, labeling and repalletizing of cardboard boxes integrating an ABB IRB 760 articulated arm industrial robot from the ABB company. To complete the diploma project, I realized the 3D simulation of the entire robotic cell using a microcontroller from the ABB company, internet networks and signals created in the ABB Robot Studio software. I made the simulation using python and C# scripts. We also made a matrix of accelerations and rotations for the integrated robot, calculations of tangential and centrifugal forces made in Matlab and an automatic tracking system made in python.

I am currently a master's student at the Polytechnic University of Bucharest, at the Faculty of Automation and Computer Science, majoring in "Computer Science, Advanced Control and Real-Time Systems", the grade obtained in the entrance exam being 10/10. I have already started the dissertation project, continuing the bachelor's project. This project involves the realization of the physically articulated arm type robot, the components being made through the additive manufacturing process (3D printing), integrating Arduino components and servomotors. For this robot, I started programming two offline and online applications. The programming of the industrial robot is done through the Arduino IDE and the C programming language, the robot having commands that it can take from the serial bus. The offline application is made with the Python programming language, having several panels that can interact with the operating system and CAD-type programs, tracking signals, positions that can be transposed in graphic form, control of individual servomotors and the control carried out on coordinate axes correlating several servomotors to achieve the movement. I have also integrated an online application, an application created as a back-end using the Python programming language, the Flask library, Jinja, pyserial (for transmitting serial signals to the robot), mathplotlib (for making graphs), pandas, numpy (for reading a document with a code already registered for programming the robot), SQL and pylogin (for the registration in a database of each account created on the online application), various APIs for webscraping of information displayed in real time from (Google search , google scholar, youtube, google patents), BS4 and Selenium for webdriver automation and information retrieval through webscraping. Also, the front-end part is made through HTML, CSS, SASS, JS, Bootstrap5, MediaQuery. The application also has an authentication and creation system for user accounts and a 2-factor authentication system, through which the user enters the email address and receives an automatic verification code.

I currently work as a System Engineer in the Continental Automotive company, ADAS Autonomus Mobility department. The technologies used are C, C++, Python, CAN, SPI, ETHERNET. On the requirements side, I use IMS, Jira, Confluence, DOORS NG, Jazz. I am assigned to 3 different projects, 50 percent I work on the "Market Intelligence and Concept Development" project, it involves the analysis of the global market and products for the improvement of Continental products, the creation of scripts with trend and seasonal data prediction algorithms, the creation system architecture (hardware and software), extracting data from the Internet and filtering them through webscraping technology and creating relevant charts for market analysis. All this is done through Python, C and Matlab languages.

For the rest of the projects, I am assigned 25 percent each, the first one being "camera, ultrasonic and satellites", realizing the software and hardware architecture together with the domains, and the other project being (DeepDrive) for which I analyze the image from the camera at the pixel level, detection, distances with Python and C.

In the past I worked as a Software Test Automation Engineer in the Luxoft company for the creation of a camera for detection, processing and transmission of signals to the ECU. I analyze exports from the script based on a test plan. The csv exports were the values of all CAN and SPI signals for each frame. Also, with each software release, we inject new software by flashing the camera, test and realize certain events in Canoe and create automatic testing scripts, automatic flashing.

Statement of Motivation
_____________________________________________________________________________________
My motivation comes from my passion for the software field, all my hobbies are related to this field, I have always tried to occupy my time developing my programming knowledge, I constantly follow courses for programming, computer science, I constantly do problems on leetcode for C , JS and Python to practice my logical thinking and develop a lot.

I really want to be accepted in the Google Summer of Code program because I want to develop myself, I want to learn as much as possible to have a very solid portfolio of projects and experience and to be able to apply all the knowledge I have. I am very excited because I have this opportunity, I want to be the best in what I do and satisfy my boundless passion for software and technology.

The programming languages and technologies I am passionate about are python, django, docker, angularjs, aws, machine learning, artificial intelligence, computer vision, deep learning, AI/ML. I want to be part of this organization because I really like the technologies you use, I want to develop myself in this part of machine learning and artificial intelligence, to be able to interact with as many experts as possible and to be able to participate in the development of an important project in this organization.

Following this organization and this project, I want to accumulate a lot of experience, learn a lot and interact with technologies and projects and be able to interact with the thinking of experts in this field.
Experiences
_____________________________________________________________________________________

My experience in this direction is accumulated both from the jobs I had and from the educational institutions I attend and from which I graduated.
Constantly, since high school, I followed a lot of trainings on programming, computer science, machine learning, artificial intelligence, but also a lot of tests for certifications.

Courses:
- Udemy - Course - 100 Days of Code: The Complete Python Pro Bootcamp
- Udemy - Course - Python Programming
- Udemy - Course - The Python Bible
- Udemy - Course - C++
- Udemy – Course – C Programming for beginners – Master the C language (in progress)
- CS50 Introduction to Computer Science – Harvard University (in progress)
- Luxoft course – Python Basic
- Luxoft course – Professional Python
- Luxoft course – Introduction to Git
- Luxoft course – Advanced C programming pointers
- Luxoft course – C++ High Performance
- Luxoft course – Open Source Software
- Luxoft course – Software Test Design
- Luxoft course – ASPICE/DQMS

Certifications:
- freeCodeCamp - Scientific Computing with Python
- HIL Academy – Test Automation
- C# -  LinkedIn Skill Assessment badge
- C (Programming Language) -  LinkedIn Skill Assessment badge
- Machine Learning -  LinkedIn Skill Assessment badge
- IT Operations - LinkedIn Skill Assessment badge
- MATLAB -  LinkedIn Skill Assessment badge
- Object-Oriented Programming (OOP) -  LinkedIn Skill Assessment badge
- Agile Methodologies -  LinkedIn Skill Assessment badge
- Git -  LinkedIn Skill Assessment badge
- Python (Programming Language) -  LinkedIn Skill Assessment badge
- ISTQB Certified Tester Foundation
- AUTOSAR Classic Fundamentals
Project details
_____________________________________________________________________________________
I would like to be part of a project that uses an advanced technology because I really like challenges, I want to learn as much as possible to interact as much as possible with advanced algorithms and structures and to interact with an advanced mentor and expert in the field. I would really like to be able to talk to an expert to choose the best possible project that will represent the best possible experience for me.

Regarding the time I want to allocate, I want to get involved in the development of the project and to work every day from Monday to Sunday and during work hours and after work hours, of course also on weekends. Time is not a problem for me, I want to devote myself as much as possible and learn as many things as possible because, as I said, I am passionate.

**Comment by thibaudcolas** [comment]
Hi @marinicadenisandrei, please take a look at our [contributor guide](https://github.com/wagtail/gsoc/blob/main/contributor-guide.md) which explains where and how we expect people to introduce themselves, among other things. We have a lot of people interested in participating to GSoC with us, in the future I would recommend to try and follow our guidelines more closely, and keep any communications concise until the actual application.
